test/test_enzyme.jl

```diff
@@ -1,13 +1,78 @@
 using Oceananigans
+using Oceananigans.TurbulenceClosures: with_tracers
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Fields: ConstantField
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: tracernames
 using Enzyme
 
 # Required presently
 Enzyme.API.runtimeActivity!(true)
-
-EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.API.looseTypeAnalysis!(true)
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Clock}) = true
+Enzyme.EnzymeRules.inactive_noinl(::typeof(Core._compute_sparams), args...) = nothing
 
 f(grid) = CenterField(grid)
 
+const maximum_diffusivity = 100
+
+"""
+    set_diffusivity!(model, diffusivity)
+
+Change diffusivity of model to `diffusivity`.
+"""
+function set_diffusivity!(model, diffusivity)
+    closure = VerticalScalarDiffusivity(; κ=diffusivity)
+    names = tuple(:c) # tracernames(model.tracers)
+    closure = with_tracers(names, closure)
+    model.closure = closure
+    return nothing
+end
+
+function set_initial_condition!(model, amplitude)
+    # Set initial condition
+    amplitude = Ref(amplitude)
+
+    # This has a "width" of 0.1
+    cᵢ(x, y, z) = amplitude[] * exp(-z^2 / 0.02 - (x^2 + y^2) / 0.05)
+    set!(model, c=cᵢ)
+
+    return nothing
+end
+
+function stable_diffusion!(model, amplitude, diffusivity)
+    set_diffusivity!(model, diffusivity)
+    set_initial_condition!(model, amplitude)
+    
+    # Do time-stepping
+    Nx, Ny, Nz = size(model.grid)
+    κ_max = maximum_diffusivity
+    Δz = 2π / Nz
+    Δt = 1e-1 * Δz^2 / κ_max
+
+    model.clock.time = 0
+    model.clock.iteration = 0
+
+    for n = 1:10
+        time_step!(model, Δt; euler=true)
+    end
+
+    # Compute scalar metric
+    c = model.tracers.c
+
+    # Hard way
```

Comment:
why this is "hard" way? to my eyes it seems "easier"?
perhaps elaborate a bit more on the comment so it's clearer?

Reply:
It's harder for AD - Enzyme can return errors with the sum function. I'm adding that to the code comment to make it clearer.

Reply:
Great!